### DISCUSSION TITLE
Get isStale using queryClient

### DISCUSSION THREAD

**Discussion by mattsputnikdigital** [TanStack/query#6714] [General]
Im struggling to find out how to use the queryClient to find out if the data is stale.

The reason I am using the queryClient is that I am using the react router loader and I want to be able to decide if to update the data if it is stale without having the <Await> component reshow the loading spinner.

**Comment by TkDodo** [comment]
> The reason I am using the queryClient is that I am using the react router loader and I want to be able to decide if to update the date if it is stale without having the component reshow the loading spinner.

`queryClient.fetchQuery` respects the `staleTime` you pass to it and will give you data from the cache if it's considered fresh

**Reply by mattsputnikdigital** [reply]
@TkDodo that is what I was using, but unfortunately if not it returns a promise and the <await> component then shows the spinner.

I would need to know if the data is stale so I could get it to do and update of the data in the background and then have it rerender with the new data rather than showing the loading spinner.

**Reply by TkDodo** [reply]
sorry, I don't get it. The default behaviour of react query is to update stale data in the background without showing a spinner ...
